feat(baseline): include data for discouraged features

LeoMcA opened the pull request:
Part of: https://github.com/mdn/fred/issues/654

Allows data for discouraged features to be included in our index.jsons, while maintaining the same asterisk logic as before: a discouraged feature is treated as having no baseline status. Without the asterisk logic change, the [font page](https://developer.mozilla.org/en-US/docs/Web/CSS/font) loses its asterisk.

LeoMcA committed:
fix(baseline): include data for discouraged features

LeoMcA committed:
feat(baseline): add urls for discouraged alternatives

LeoMcA committed:
Merge remote-tracking branch 'origin/main' into baseline-discouraged

LeoMcA committed:
Merge remote-tracking branch 'origin/main' into baseline-discouraged

LeoMcA committed:
Merge remote-tracking branch 'origin/main' into baseline-discouraged

LeoMcA committed:
feat(baseline): add reason_html and removal_date to discouraged data

LeoMcA committed:
chore: run cargo fmt

caugner commented on crates/rari-data/src/feature_urls.rs:
```diff
@@ -0,0 +1,44 @@
+
+static FEATURE_MDN_URLS: LazyLock<IndexMap<&'static str, &'static str>> = LazyLock::new(|| {
+    [
+        ("async-clipboard", "/docs/Web/API/Clipboard_API"),
```
Would be nice if web-features could add an `mdn` field: https://github.com/web-platform-dx/web-features/blob/96d7e3c4e91500cf84f31cbab590f539566980ae/features/async-clipboard.yml#L1-L5

LeoMcA replied:
We've been talking about how best to maintain the web-feature <--> mdn mapping: once we've decided (and implemented) that, I'm sure we'll be able to add an mdn field to web-features.

Perhaps for these "features pointed at by discouraged features" it makes sense to add them manually in the web-features repo though

LeoMcA committed:
Merge remote-tracking branch 'origin/main' into baseline-discouraged

LeoMcA committed:
add tests covering missing urls and missing features for alternatives

LeoMcA committed:
map feature ids to mdn urls

LeoMcA committed:
explicitly exclude discouraged features from the baseline sub_status aggregation

this ensures that when we stop ignoring discouraged features, we don't change asterisk computation

LeoMcA committed:
resolve discouraged features

LeoMcA committed:
add alternatives to the Baseline struct and resolve

LeoMcA committed:
add tests to cover current asterisk behaviour

fix bug where subkeys of last sorted bcd key were ignored

LeoMcA committed:
expose reason_html and removal_date from discouraged data

globau commented on crates/rari-data/src/baseline.rs:
```diff
@@ -185,9 +202,38 @@ impl WebFeatures {
+                    })
+                    .collect(),
+                None => Vec::new(),
+            };
```
`self.feature_data_by_name(alternative)` returns `None` when the alternative key doesn't exist in the web-features data - we should log a warning in this case.  Current logic won't generate any warnings.

globau reviewed (approved):
lgtm, just a minor missing warning

LeoMcA committed:
add warning for missing alternative in web features data
